### PR TITLE
Feature/mactabviewcontroller

### PR DESCRIPTION
**Pull request by tofutim**
### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)
Allows binding the SelectedTabViewItemIndex in NSTabViewController

### :arrow_heading_down: What is the current behavior?
Can't do it, nothing happens

### :new: What is the new behavior (if this is a feature change)?
Can now bind SelectedTabViewItemIndex to ints and enums

### :boom: Does this PR introduce a breaking change?
No

### :bug: Recommendations for testing
See Playground.Mac

### :memo: Links to relevant issues/docs
https://github.com/MvvmCross/MvvmCross/issues/2191

### :thinking: Checklist before submitting

- [x] All projects build
- [x] Follows style guide lines ([code style guide](https://github.com/MvvmCross/MvvmCross#code-style-guidelines))
- [x] Relevant documentation was updated ([docs style guide](https://www.mvvmcross.com/documentation/contribute/mvvmcross-docs-style-guide))
- [x] Nuspec files were updated (when applicable)
- [x] Rebased onto current develop

### CODE REVIEW

File: MvvmCross/Binding/Mac/Target/MvxNSTabViewControllerSelectedTabViewItemIndexTargetBinding.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        public override void SubscribeToEvents()
+        {
+            var view = View;
+            if (view == null)
+            {
+                MvxBindingTrace.Trace(MvxTraceLevel.Error, "Error - NSTabViewController is null in MvxNSTabViewControllerSelectedTabViewItemIndexTargetBinding");
+                return;
+            }
+
+            this._subscribed = true;
```
Comment (nmilcoff) [inline MvvmCross/Binding/Mac/Target/MvxNSTabViewControllerSelectedTabViewItemIndexTargetBinding.cs]: Can you please remove the `this` keyword here?

File: MvvmCross/Binding/Mac/Target/MvxNSTabViewControllerSelectedTabViewItemIndexTargetBinding.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            view.SelectedTabViewItemIndex = (int)value;
+        }
+
+        protected override void Dispose(bool isDisposing)
+        {
+            base.Dispose(isDisposing);
+            if (isDisposing)
+            {
+                var view = View;
+                if (view != null && this._subscribed)
```
Comment (nmilcoff) [inline MvvmCross/Binding/Mac/Target/MvxNSTabViewControllerSelectedTabViewItemIndexTargetBinding.cs]: Same with `this` here

File: MvvmCross/Binding/Mac/Target/MvxNSTabViewControllerSelectedTabViewItemIndexTargetBinding.cs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+                        try
+                        {
+                            view.TabView.DidSelect -= HandleValueChanged;
+                        }
+                        catch (Exception ex)
+                        {
+                            MvxBindingTrace.Error(ex.Message);
+                        }
+                    }
+                    this._subscribed = false;
```
Comment (nmilcoff) [inline MvvmCross/Binding/Mac/Target/MvxNSTabViewControllerSelectedTabViewItemIndexTargetBinding.cs]: Same with `this` here

**Comment by tofutim** [comment]
Ok, I think I got everything. I leave to you guys!

**Pull request merged**